tests/unit/test_build_rules.py

```diff
@@ -1005,3 +1005,218 @@ def test_dataset_rule_null_columns_items_in_kwargs():
                 "limit": 1,
             },
         )
+
```

Comment:
Negative tests are missing for raising TypeError or unsupported column expressions such as  `F.col("val") + F.lit(1)`